compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -210,15 +210,21 @@ internal class BindingModuleGenerator(
           .toList()
       }
 
-      val generatedMethods = getContributedBindingClasses(
-        collectedClasses = contributedBindingClasses,
-        annotationFqName = contributesBindingFqName,
-        isMultibinding = false
-      ) + getContributedBindingClasses(
-        collectedClasses = contributedMultibindingClasses,
-        annotationFqName = contributesMultibindingFqName,
-        isMultibinding = true
-      )
+      val generatedMethods =
+        getContributedBindingClasses(
+          collectedClasses = contributedBindingClasses,
+          annotationFqName = contributesBindingFqName,
+          isMultibinding = false
+        )
+          .plus(
+            getContributedBindingClasses(
+              collectedClasses = contributedMultibindingClasses,
+              annotationFqName = contributesMultibindingFqName,
+              isMultibinding = true
+            )
+          )
+          .renameDuplicateFunctions()
+          .sorted()
```

Comment:
I think it was a problem that they never were sorted.

Reply:
> I think it was a problem that they never were sorted.

Can you clarify why that was a problem? Vs just renaming the duplicate functions

Reply:
For stable caching it's important to produce the same output the for the same input. I'm honestly not sure right now if it has an impact in practice, but it's always better to have a stable sort.